sdk/swap-service/src/implementation/oneinch/OneInchSwapProvider.ts

```diff
@@ -0,0 +1,92 @@
+  public type: SwapProviderType = SwapProviderType.OneInch
+
+  private readonly _apiUrl: string
+  private readonly _version: string
+  private readonly _allowedSwapProtocols: string[]
+  private readonly _apiKey: string
+
+  constructor(params: OneInchSwapProviderConfig) {
+    this._apiUrl = params.apiUrl
+    this._version = params.version
+    this._apiKey = params.apiKey
+    this._allowedSwapProtocols = params.allowedSwapProtocols
+  }
+
+  public async getSwapData(params: {
+    chainInfo: ChainInfo
+    fromAmount: TokenAmount
+    toMinimumAmount: TokenAmount
+    recipient: Address
+    slippage: Percentage
+  }): Promise<SwapData> {
+    const swapUrl = this._formatOneInchSwapUrl({
+      chainInfo: params.chainInfo,
+      fromTokenAmount: params.fromAmount,
+      toMinimumAmount: params.toMinimumAmount,
+      recipient: params.recipient,
+      slippage: params.slippage,
+    })
+
+    const authHeader = this._getOneInchAuthHeader()
+
+    const response = await fetch(swapUrl, {
+      headers: authHeader,
+    })
+
+    if (!(response.status === 200 && response.statusText === 'OK')) {
+      throw new Error(`Error performing 1inch swap request ${swapUrl}: ${await response.body}`)
+    }
+
+    const responseData = (await response.json()) as OneInchSwapResponse
+
+    return {
+      provider: SwapProviderType.OneInch,
+      fromTokenAmount: params.fromAmount,
+      toTokenAmount: params.toMinimumAmount,
+      calldata: responseData.tx.data as Hex,
+      targetContract: Address.createFrom({ hexValue: responseData.tx.to }),
+      value: responseData.tx.value,
+      gasPrice: responseData.tx.gasPrice,
+    }
+  }
+
+  private _getOneInchAuthHeader(): OneInchAuthHeader {
+    return { [OneInchAuthHeaderKey]: this._apiKey }
+  }
+
+  private _formatOneInchSwapUrl(params: {
+    chainInfo: ChainInfo
+    fromTokenAmount: TokenAmount
+    toMinimumAmount: TokenAmount
+    recipient: Address
+    slippage: Percentage
+    disableEstimate?: boolean
+    allowPartialFill?: boolean
+  }): string {
+    const chainId = params.chainInfo.chainId
+    const fromTokenAddress = params.fromTokenAmount.token.address.toString().toLowerCase()
+    const toTokenAddress = params.toMinimumAmount.token.address.toString().toLowerCase()
+    const fromAmount = params.fromTokenAmount.toWei()
+    const recipient = params.recipient.toString()
+    const disableEstimate = params.disableEstimate ? params.disableEstimate : true
+    const allowPartialFill = params.allowPartialFill ? params.allowPartialFill : false
+    const protocolsParam = this._allowedSwapProtocols.length
+      ? this._allowedSwapProtocols.join(',')
+      : ''
+
+    return `${this._apiUrl}/${this._version}/${chainId}/swap?fromTokenAddress=${fromTokenAddress}&toTokenAddress=${toTokenAddress}&amount=${fromAmount}&fromAddress=${recipient}&slippage=${params.slippage.toString()}&protocols=${protocolsParam}&disableEstimate=${disableEstimate}&allowPartialFill=${allowPartialFill}`
+  }
```

Comment:
<img width="710" alt="image" src="https://github.com/OasisDEX/summerfi-monorepo/assets/15217169/c97e97f8-7892-4194-b225-c2187501df15">

Strage, it looks like they have changed the api (again) there is no longer `fromTokenAddress` and `toTokenAddress`  to `src` and `dist` but former still work. Probably in future we will need to implement new providers based on one inch version.

Reply:
Good to know! Yeah, now that the provider is abstracted properly it should be easier to change it to adapt it to other versions